src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -0,0 +1,170 @@
+  * @param bases_total the total number of bases observed at this position across all reads.
+  */
+case class ErrorRateByReadPositionMetric
+( position: Int,
+  error_rate_read_one: Double,
+  error_rate_read_two: Double,
+  error_rate_total: Double,
+  bases_read_one: Long,
+  bases_read_two: Long,
+  bases_total: Long
+) extends Metric
+
+private class ErrorRateByReadPositionCounter {
+  val r1TotalBasesByReadPosition = new NumericCounter[Int]()
+  val r1ErrorBasesByReadPosition = new NumericCounter[Int]()
+  val r2TotalBasesByReadPosition = new NumericCounter[Int]()
+  val r2ErrorBasesByReadPosition = new NumericCounter[Int]()
+
+  def accept(rec: SAMRecord): Unit = {
+    val first: Boolean    = !rec.getReadPairedFlag || rec.getFirstOfPairFlag
+    val totalBasesByReadPosition = if (first) r1TotalBasesByReadPosition else r2TotalBasesByReadPosition
+    val errorBasesByReadPosition = if (first) r1ErrorBasesByReadPosition else r2ErrorBasesByReadPosition
+    val read              = maybeReverseComplement(rec.getReadBases.clone(), rc=rec.getReadNegativeStrandFlag)
+    val reference         = maybeReverseComplement(SequenceUtil.makeReferenceFromAlignment(rec, false), rc=rec.getReadNegativeStrandFlag)
+
+    var readPosition = 0
+    while (readPosition < read.length) {
+      val refBase  = reference(readPosition)
+      val readBase = read(readPosition)
+      // Make sure we are not at an insertion and not soft-clipped.  Also the bases must be valid
+      if ('-' != refBase && '0' != refBase && validBase(refBase) && validBase(readBase)) {
+        if (!SequenceUtil.basesEqual(read(readPosition), reference(readPosition))) { // mismatch
+          errorBasesByReadPosition.count(readPosition)
+        }
+        totalBasesByReadPosition.count(readPosition)
+      }
+      readPosition += 1
+    }
+  }
+
+  def metrics: Seq[ErrorRateByReadPositionMetric] = {
+    if (r1TotalBasesByReadPosition.isEmpty && r2TotalBasesByReadPosition.isEmpty) return Seq.empty
+    // create the metrics
+    val maxReadPosition = (r1TotalBasesByReadPosition.map(_._1) ++ r2TotalBasesByReadPosition.map(_._1)).max
+    Seq.range(0, maxReadPosition+1).map { readPosition =>
+      val r1Error = r1ErrorBasesByReadPosition.countOf(readPosition)
+      val r1Total = r1TotalBasesByReadPosition.countOf(readPosition)
+      val r2Error = r2ErrorBasesByReadPosition.countOf(readPosition)
+      val r2Total = r2TotalBasesByReadPosition.countOf(readPosition)
+
+      new ErrorRateByReadPositionMetric(
+        position = readPosition + 1,
+        error_rate_read_one = divide(r1Error, r1Total),
+        error_rate_read_two = divide(r2Error, r2Total),
+        error_rate_total    = divide(r1Error + r2Error, r1Total + r2Total),
+        bases_read_one      = r1Total,
+        bases_read_two      = r2Total,
+        bases_total         = r1Total + r2Total
+      )
+    }
+  }
+
+  private def divide(num: Long, denom: Long): Double = {
```

Comment:
Just to be safe, num should also be zero any time denom is zero right?  Could be written as:
```scala
private def divide(num: Long, denom: Long): Double = (num, denom) match {
  case (0,0) => 0
  case (n,d) => n / d.toDouble
}
```
since I know you like matches 😄